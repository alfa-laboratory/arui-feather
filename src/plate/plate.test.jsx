/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import { shallow, mount } from 'enzyme';

import Plate from './plate';

describe('plate', () => {
    it('should render without problems', () => {
        let plate = shallow(<Plate>plate-text</Plate>);

        expect(plate).toMatchSnapshot();
        expect(plate.html()).toContain('plate-text');
    });

    it('should render with cross without problems', () => {
        let plate = shallow(<Plate hasCloser={ true }>plate-text</Plate>);
        let crossNode = plate.find('.plate__closer');

        expect(crossNode).toBeTruthy();
    });

    it('should call `onCloserClick` callback after plate closer was clicked', () => {
        let onCloserClick = jest.fn();
        let plate = shallow(
            <Plate hasCloser={ true } onCloserClick={ onCloserClick }>plate-text</Plate>
        );
        let closer = plate.find('.plate__closer');

        closer.simulate('click');

        expect(onCloserClick).toHaveBeenCalled();
    });

    it('should call `onClick` callback after plate was clicked', () => {
        let onClick = jest.fn();
        let plate = mount(
            <Plate hasCloser={ true } onClick={ onClick }>plate-text</Plate>
        );

        plate.simulate('click');

        expect(onClick).toHaveBeenCalled();
    });
});
