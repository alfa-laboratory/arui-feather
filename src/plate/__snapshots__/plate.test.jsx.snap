// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plate should render without problems 1`] = `
<ThemeProvider
  theme="alfa-on-white"
>
  <div
    className="plate plate_theme_alfa-on-white"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <div
      className="plate__content"
    >
      plate-text
    </div>
  </div>
</ThemeProvider>
`;
